src/graphics/Graphics.hpp

```diff
@@ -9,13 +9,123 @@
 #ifndef GRAPHICS_HPP
 #define GRAPHICS_HPP
 
+#include <SDL2/SDL.h>
+#include <vector>
+#include <memory>
+#include "../common/IScene.hpp"
+#include "../common/IObject.hpp"
 
-
+/**
+ * @class Graphics
+ * @brief This class handles all graphics-related operations.
+ * @details The Graphics class is responsible for initializing and managing the rendering
+ *          of objects in the scenes, managing an SDL window and renderer, and ensuring 
+ *          objects are sorted by their z-index for proper rendering order.
+ * @version v0.1.0
+ * @since v0.1.0
+ * @author Aubane NOURRY
+ */
 class Graphics {
 public:
-    Graphics();
-};
+    /**
+     * @brief Constructor for the Graphics class.
+     * @param width The width of the window.
+     * @param height The height of the window.
+     * @param title The title of the window. Default is "Game".
+     * @param precharge Whether to preload all objects in all scenes. Default is false.
+     * @details This initializes the window with the specified dimensions and title,
+     *          and optionally precharges all objects if the precharge flag is set to true.
+     * @since v0.1.0
+     * @author Aubane NOURRY
+     */
+    Graphics(int width, int height, const char *title = "Game", bool precharge = false);
+
+    /**
+     * @brief Destructor for the Graphics class.
+     * @details Cleans up and releases all resources used by the Graphics class, including
+     *          the SDL window and renderer.
+     * @since v0.1.0
+     * @author Aubane NOURRY
+     */
+    ~Graphics();
+
+    /**
+     * @brief Initializes the SDL window and renderer.
+     * @details This method initializes the SDL window and renderer based on the provided 
+     *          width, height, and title. It should be called before any rendering operations.
+     * @since v0.1.0
+     * @author Aubane NOURRY
+     */
+    void initialize();
```

Comment:
Not a big fan of the name 'initialize', if it's called (and I quote) "before any rendering operations" I don't think initialize is the good term since it implies a one time operation. Maybe "prepare" would be a better one, but I really don't see why anyone should could a function before doing anything. Can you elaborate on this ?